app/models/patient.rb

```diff
@@ -162,6 +164,33 @@ def event_params
     }
   end
 
+  def confirm_unique_phone_number
+    ##
+    # This method is preferred over Rail's built-in uniqueness validator
+    # so that case managers get a meaningful error message when a patient
+    # exists on a different line than the one the volunteer is serving.
+    #
+    # See https://github.com/DCAFEngineering/dcaf_case_management/issues/825
+    ##
+    # phones are unique, so there should be zero or one match
+    match = Patient.where(primary_phone: primary_phone).first
```

Comment:
Ah, I see what you're doing! I think there a couple ways we can tidy this up.

I'd suggest renaming this variable to something that isn't a predefined method name in the ruby base library -- it should work fine, but this can be a total Zone of Pain kinda thing. Might suggest renaming it to 'phone_match' or something like that.

`.where` methods in activerecord/mongoid return an enumerable, as you've correctly noted with your `.first` here. You might be able to simplify this by changing it to a `.find`, which will either return nil or the single object.

You could also consider rewriting the query a little bit to explicitly exclude the patient's id -- it'll simplify your guard clause a little bit. Maybe something like the find version of `Patient.where.not(id: id.to_s)`, or the more mongoid version (I THINK) of `Patient.find_by(:id.ne => id.to_s)`? (you'll probably want to cast it to a string because the patient object considers it a BSON id, but it might work out of the box.)

In short, you can probably collapse these four lines into something similar to:

```
return true if Patient.find_by(primary_phone: primary_phone, :id.ne => id.to_s)
```

Don't feel married to this if it doesn't work, by the way! I'm pretty sleep deprived and you've spent more time thinking about this than I have!